nested-spire/README.md

```diff
@@ -0,0 +1,210 @@
+The other point to highlight is the `-downstream` option. This option when set, indicates that the entry describes a downstream SPIRE server.
+
+## Run the Scenario
+
+Use the `set-env.sh` script to run all the services that compound the scenario. All the configurations described in the previous sections are illustrated in the different services.
+Ensure that the current working directory is `.../spire-tutorials/nested-spire` and run:
+
+```console
+    bash scripts/set-env.sh
+```
+
+Once the script is completed, in another terminal run the following command to review the logs from all the services
+
+```console
+    docker-compose logs -f -t
+```
+
+
+# Part 2: Test the Deployments
+
+Now that the SPIRE deployments are ready, let's test the scenario that we've configured.
+
+## Create Workload Registration Entries
+
+To test the scenario we create two workload registration entries, one entry per each of the nested SPIRE deployment. The goal of the test case is to demonstrate that SVIDs created in a nested configuration are valid in the entire trust domain, not only in the scope of the server that originated the SVID.
+
+```console
+   # Workload for nestedA deployment
+   docker-compose exec -T nestedA-server \
+       /opt/spire/bin/spire-server entry create \
+       -parentID "spiffe://example.org/spire/agent/x509pop/$(fingerprint nestedA/agent/agent.crt.pem)" \
+       -spiffeID "spiffe://example.org/nestedA/workload" \
+       -selector "unix:uid:1001" \
+       -ttl 0
+
+   # Workload for nestedB deployment
+   docker-compose exec -T nestedB-server \
+       /opt/spire/bin/spire-server entry create \
+       -parentID "spiffe://example.org/spire/agent/x509pop/$(fingerprint nestedB/agent/agent.crt.pem)" \
+       -spiffeID "spiffe://example.org/nestedB/workload" \
+       -selector "unix:uid:1001" \
+       -ttl 0
+```
+
+The examples uses the `$(fingerprint nestedA/agent/agent.crt.pem)` notation again to show that the `-parentID` flag is set to the SPIFFE ID of the nested SPIRE Agent. TTL equals to zero indicates that the default value of 3600 seconds will be used. Finally, in both cases there is an unix selector to identify any process that is run by the user with id 1001.
+Use the following Bash script to create the registration entries:
+
+```console
+   bash scripts/create-workload-registration-entries.sh
+```
+
+## Run the Test
+
+Once both workload registration entries are propagated, let's test that SVIDs created in a nested configuration are valid in the entire trust domain, not only in the scope of the server that originated the SVID.
+
+The test consists on getting a JWT-SVID from the `nestedA-agent` and validate it using the `nestedB-agent`. In both cases, Docker Compose run the processes using the user 1001 to match the workload registration entries created in the previous section.
+
```

Comment:
I think we can run these commands with the test.sh script, right?

```suggestion
The `test.sh` script runs the following commands:
```

Reply:
Yes, test.sh run the same commands to check the configuration is correct.
I didn't use the script intentionally to force the reader to run it manually as part of the tutorial the same way it is done with the other tutorials in the repository.
I think test.sh is something internal and because of that I never mention it.

That said, I'm open to change it if you consider it a better approach.

Reply:
That's an excellent point, I agree with you. We shouldn't use test.sh but instead show the actual commands.

Reply:
Good. I'll discard this suggestion and also the next one that is linked to this comment. 
I'll mark both as resolved.